### PR TITLE
SAAS-2298: Plugin to check sender policy FW record exists in Route 53 DNS records

### PR DESCRIPTION
**Pull request by UmairAjmal**

### CODE REVIEW

File: plugins/aws/route53/senderPolicyFwInUse.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Sender Policy Framework In Use',
+    category: 'Route53',
+    description: 'Ensure that Sender Policy Framework (SPF) is used to stop spammers from spoofing your AWS Route 53 domain.',
+    more_info: 'Implementing SPF records for Route 53 domain names will help in reducing spam and increasing domains trustworthiness by stopping email spoofing.',
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwInUse.js:8]:
```suggestion
    more_info: 'Implementing SPF records for Route53 domain names will help in reducing spam and increasing domains trustworthiness by stopping email spoofing.',
```

File: plugins/aws/route53/senderPolicyFwInUse.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            return callback(null, results, source);
+        }
+
+        if (!listHostedZones.data.length) {
+            helpers.addResult(results, 0, 'No Route53 Hosted Zones found', region);
+            return callback(null, results, source);
+        }
+
+        async.each(listHostedZones.data, function(zone, cb){
+            var resource = `arn:aws:route53:::${zone.Id}`;
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwInUse.js:36]: add check if (!zone.id) return cb();

File: plugins/aws/route53/senderPolicyFwInUse.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            if (!listResourceRecordSets.data.ResourceRecordSets || !listResourceRecordSets.data.ResourceRecordSets.length) {
+                helpers.addResult(results, 0,
+                    'No resource record sets found',
+                    region, resource);
+                return cb();
+            }
+
+            var spfDisabledDnsRecords = [];
+            listResourceRecordSets.data.ResourceRecordSets.forEach(recordSet => {
+                if (recordSet.Type && recordSet.Type === 'TXT' && recordSet.ResourceRecords && recordSet.ResourceRecords.length) {
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwInUse.js:57]:
```suggestion
                if (recordSet.Type && recordSet.Type.toUpperCase() === 'TXT' && recordSet.ResourceRecords && recordSet.ResourceRecords.length) {
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/route53/senderPolicyFwInUse.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Sender Policy Framework In Use',
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwInUse.js:5]:
```suggestion
    title: 'Sender Policy Framework Record Present',
```

File: plugins/aws/route53/senderPolicyFwInUse.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Sender Policy Framework Record Present',
+    category: 'Route53',
+    description: 'Ensure that Sender Policy Framework (SPF) is used to stop spammers from spoofing your AWS Route 53 domain.',
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwInUse.js:7]: Look into description, more_info and link

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/route53/senderPolicyFwInUse.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    more_info: 'Implementing SPF records for Route53 domain names will help in reducing spam and increasing domains trustworthiness by stopping email spoofing.',
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwInUse.js:8]:
```suggestion
    more_info: 'Implementing SPF records for Route 53 domain names will help in reducing spam and increasing domains trustworthiness by stopping email spoofing.',
```

File: plugins/aws/route53/senderPolicyFwRecordPresent.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+        if (!listHostedZones.data.length) {
+            helpers.addResult(results, 0, 'No Route53 Hosted Zones found', region);
+            return callback(null, results, source);
+        }
+
+        async.each(listHostedZones.data, function(zone, cb){
+            var resource = `arn:aws:route53:::${zone.Id}`;
+
+            var listResourceRecordSets = helpers.addSource(cache, source,
```
Comment (AkhtarAmir) [inline plugins/aws/route53/senderPolicyFwRecordPresent.js:38]:
```suggestion
        async.each(listHostedZones.data, function(zone, cb){
            if (!zone.Id) return cb();

            var resource = `arn:aws:route53:::${zone.Id}`;

            var listResourceRecordSets = helpers.addSource(cache, source,
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**